src/hw_stats.py

```diff
@@ -33,16 +32,19 @@
         'core_control':param.ccu_pow,
         'tile_control':param.tcu_pow
         }
+# 'adc':param.adc_pow_dyn, 
+for k in range (cfg.num_adc):
+    hw_comp_energy['adc_'+str(k)] = param.adc_pow_dyn_dict[str(cfg.adc_res_new[str('matrix_adc_'+str(k))])]
```

Comment:
Can `str(cfg.adc_res_new[str('matrix_adc_'+str(k))])` also be put within `hetero-adc` flag, so that in cases where `num_matrix>2`, this can fall back to `cfg.adc_res` ?

Reply:
Please look my suggestion above